feat: download guardian backup

Kodylow opened the pull request:
Bumps flake to most recent fedimint master commit to get the download guardian config endpoint and adds flow to download backup: button next to Federation Config + modal to acknowledge they're downloading private key material

otech47 commented on apps/guardian-ui/src/languages/en.json:
```diff
@@ -55,7 +55,14 @@
+      "downloadBackup": {
+        "button": "Download Backup",
+        "warningTitle": "Warning",
+        "warningText": "The backup contains private keys and secret material for the federation guardian and must be kept secure.\n Recovery using this backup requires your admin password!",
```
didn't test this but from my understanding of the `useTranslation` hook in other projects, line breaks won't work out of the box like this. might need to render it in a `<Trans>` component for this to do what you expect.

Kodylow replied:
just removed linebreak, isn't necessary.

Kodylow commented:
moved button and model into new DownloadBackup.tsx and moved it to the bottom of the screen, design feedback was that putting it on the Federation Config meant it look like that was the thing they were backing up not the whole guardian dir 
<img width="618" alt="image" src="https://github.com/fedimint/ui/assets/74332828/307a4418-d117-4c7c-a47e-548aca00e533">
